src/test/scala/org/scalameter/MethodInvocationCountTest.scala

```diff
@@ -0,0 +1,7 @@
+package org.scalameter
+
+class MethodInvocationCountTest extends InvocationCountMeasurerTest {
+  test("MethodInvocationCountTest.allocations should be deterministic") {
+    checkInvocationCountMeasurerTest(new org.scalameter.examples.MethodInvocationCountTest1)
```

Comment:
I wouldn't use `finalize` here - it is too brittle. What we really want is:
1. reserve a temporary file before the test ctor is called, so that context can get a new path - done by the `Measurer#prepareContext` in `BasePerformanceTest`.
2. use the temporary file to instrument the classpath just before tests are executed - done by `Measurer#beforeTests` in `executeTests`.
3. delete the temporary jar file - done by `Measurer#afterTests`.

This way, the same temporary file path is reused multiple times, but created and deleted right before and after the tests start/complete.
I think this is the right design.